python/cudf/cudf/tests/test_series.py

```diff
@@ -2640,6 +2652,26 @@ def test_series_setitem_mixed_bool_dtype():
         s[0] = 10
 
 
+@pytest.mark.parametrize(
+    "nat, value",
+    [
+        [np.datetime64("nat"), np.datetime64("2020-01-01")],
+        [np.timedelta64("nat"), np.timedelta64(1)],
+    ],
+)
+def test_series_np_array_nat_nan_as_null_false(nat, value, request):
+    expected = np.array([nat, value])
+    if expected.dtype.kind == "m":
+        request.applymarker(
+            pytest.mark.xfail(
+                raises=TypeError, reason="timedelta64 not supported by cupy"
+            )
+        )
+    ser = cudf.Series(expected, nan_as_null=False)
```

Comment:
```suggestion
)
@pytest.mark.parametrize("nan_as_null", [True, False])
def test_series_np_array_nat_nan_as_nulls(nat, value, request, nan_as_null):
    expected = np.array([nat, value])
    if expected.dtype.kind == "m":
        request.applymarker(
            pytest.mark.xfail(
                raises=TypeError, reason="timedelta64 not supported by cupy"
            )
        )
    ser = cudf.Series(expected, nan_as_null=nan_as_null)
```

Maybe better to validate both cases?

Reply:
Good idea. Will include this